include/meta/meta.hpp

```diff
@@ -2125,6 +2125,21 @@ namespace meta
             using min = defer<min, T, U>;
         }
 
+        /// Index of the first occurrence of the type \p T within the list \p List.
+        /// \par Complexity
+        /// \f$ O(N) \f$.
+        /// \ingroup list
+        template <typename T, typename List>
```

Comment:
At the very least, the docs should say what it returns when the element is not found. Do we want something like a `meta::optional`? What would such a thing look like, I wonder? That discussion leads to monads and pattern matching -- potentially very interesting stuff. I might be over-thinking it, though. ;-)